docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,196 @@
+The data structure affords incrementing and decrementing the shared value via its `increment` method. Decrements are done by providing a negative value.
+
+The `SharedCounter` is a specialized [optimistic DDS]({{< relref "dds.md#optimistic-data-structures" >}}).
+It operates on communicated _deltas_ (amounts by which the shared value should be incremented or decremented), rather than direct changes to the shared value.
+In this way, it avoids the pitfalls of DDSes with simpler merge strategies, in which one user's edit may clobber another's (see [below](#why-a-specialized-dds)).
+
+Note that the `SharedCounter` only operates on integer values.
+
+### Why a specialized DDS?
+
+You may be asking yourself, why not just store the shared integer value directly in another DDS like a [SharedMap][]?
+Why incur the overhead of another runtime type?
+
+The key to the answer here is that DDSes with simpler merge strategies (like `SharedMap`) take a somewhat brute-force approach to merging concurrent edits.
+For a semantic data type like a counter, this can result in undesirable behavior.
+
+#### SharedMap Example
+
+Let's illustrate the issue with an example.
+
+Here, we will use a [SharedMap][] to store our shared integer value.
+For simplicity, it will be stored under a static key: `counter-key`.
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+Each edit is then sequenced and broadcast to other collaborators.
+
+For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
+This is intentional.
+Since increments and decrements are [commutative](https://en.wikipedia.org/wiki/Commutative_property), it shouldn't matter in what order two nearly concurrent increment operations occur; the result _should_ be the same.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = **44**, right?
+Again, note that we are ignoring which `+ 1` came from which user, as the order should not matter.
+
+Unfortunately, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
+Regardless of which user's edit was sequenced first and which was sequenced last, each user's client will update their map entry to be 42 (the current value they see) + 1 = **43**, and the latter of the two operations to be sequenced will clobber the first.
```

Comment:
This is fine, though it's kind of less about "which order are they applied in" and more about "can my change be combined with the changes from other users, even if I haven't seen that change locally yet"

Reply:
@ChumpChief would you mind elaborating a bit? I see that being the general invariant trying to be kept, but since this is a concrete example, I wanted to give a very explicit case. Any suggestions for what you'd like to see here?

Reply:
"Since both clients saw the value 42 when the button was clicked, both clients will try to write 43 (since neither has seen the other's change yet).  With no one even trying to write 44, we can't get the right answer."

The clobbering and relative order doesn't actually make a difference in this case - instead the problem is that no one can know to write 44.

Clobbering comes into play more in the SharedMap vs. SharedString examples, where one client writes 'a' and the other writes 'b' and you want "ab" -- in this case the clobbering makes a difference.

Reply:
I see. That makes sense. Will update.

Reply:
I completely re-worked the example. The new scenario more explicitly walks the user through how the updates are applied, and I think it is much clearer.